src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,46 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo | null;
+  onResetTodo: (boolean: null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  todo,
+  onResetTodo,
+}) => {
+  const [user, setUser] = useState<User>();
+  const [isUserLoaded, setIsUserLoaded] = useState(false);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      if (todo) {
+        const currentUser = await getUser(todo.userId);
+
+        setUser(currentUser);
+      }
+
+      setIsUserLoaded(true);
+    };
+
+    fetchUser();
+  }, []);
```

Comment:
The useEffect dependency array is empty, causing it to run only once when the component mounts. Add 'todo' to the dependency array to fetch the user if 'todo' changes.